src/audio/codec_adapter/codec/passthrough.c

```diff
@@ -7,15 +7,20 @@
 // Passthrough codec implementation to demonstrate Codec Adapter API
 
 #include <sof/audio/codec_adapter/codec/generic.h>
-#include <sof/audio/codec_adapter/codec/passthrough.h>
 
-int passthrough_codec_init(struct comp_dev *dev)
+/* 0495ef10-254b-473d-8cbf-3d2292945c24 */
+DECLARE_SOF_RT_UUID("passthrough_codec", passthrough_uuid, 0x0495ef10, 0x254b, 0x473d,
```

Comment:
I don't like the passthru name here, I would stick to the adapter name as passthru suggest literally no processing. Also I miss the general picture, how are we going to approach the scenario where codecs are dynamically binded with the adapter base on their UUID?

Reply:
@mrajwa adapter base is now just some sort of library helper functions. Each codec will create their own component based on UUID. The binding you mentioned is created statically inside the code like this:

```
static const struct comp_driver comp_codec_adapter = { \
	.type = SOF_COMP_NONE, \
	.uid = SOF_RT_UUID(uuid), \
	.tctx = &tr, \
	.ops = { \
		.create = adapter_shim_new, \
		.prepare = codec_adapter_prepare, \
		.params = codec_adapter_params, \
		.copy = codec_adapter_copy, \
		.cmd = codec_adapter_cmd, \
		.trigger = codec_adapter_trigger, \
		.reset = codec_adapter_reset, \
		.free = codec_adapter_free, \
	}, \
}; \
```

Reply:
The codec adapter interface is more open to change than the component interface, therefore we can register that later in my macro and then use that to dynamically switch based on information we add into the interface. This is just a step toward using only UUIDs as identification.